src/coreclr/src/zap/zapinfo.cpp

```diff
@@ -2150,15 +2150,19 @@ DWORD FilterNamedIntrinsicMethodAttribs(ZapInfo* pZapInfo, DWORD attribs, CORINF
         // is because they often change the code they emit based on what ISAs are supported by the compiler,
         // but we don't know what the target machine will support.
         //
-        // Additionally, we make sure none of the hardware intrinsic method bodies (except ARM64) get pregenerated in crossgen
+        // Additionally, we make sure none of the hardware intrinsic method bodies get pregenerated in crossgen
         // (see ZapInfo::CompileMethod) but get JITted instead. The JITted method will have the correct
         // answer for the CPU the code is running on.
 
-        // For Arm64, AdvSimd/ArmBase is the baseline that is suported and hence we do pregenerate the method bodies
-        // of ARM64 harware intrinsic.
         fTreatAsRegularMethodCall = (fIsGetIsSupportedMethod && fIsPlatformHWIntrinsic);
-#if !defined(TARGET_ARM64)
-        fTreatAsRegularMethodCall |= (!fIsPlatformHWIntrinsic && fIsHWIntrinsic);
+
+#if defined(TARGET_ARM64)
+        if (!fIsPlatformHWIntrinsic && fIsHWIntrinsic)
+        {
+            fTreatAsRegularMethodCall = (strcmp(className, "Vector64`1") != 0) && (strcmp(className, "Vector128`1") != 0);
```

Comment:
Why don't we need something similar for `Vector128<T>` and `Vector256<T>` on x86?

Reply:
(I might just be missing where we cover it, if it exists somewhere)

Reply:
My understanding (that could be incorrect) is that on x86 in R2R code we want to treat all calls outside of System.Runtime.Intrinsics.X86 as regular calls since there is no guarantee what ISAs will be supported on target platform while on Arm64 - ArmBase and AdvSimd are baseline and always supported.

@davidwrighton Is it correct interpretation of discussion in https://github.com/dotnet/runtime/pull/38060?

Reply:
Ah right. I remember this reasoning now. It always confuses me at first 😄